packages/x-tree-view/src/useTreeItem2/useTreeItem2.ts

```diff
@@ -82,6 +81,8 @@ export const useTreeItem2 = <
   const createContentHandleClick =
     (otherHandlers: EventHandlers) => (event: React.MouseEvent & MuiCancellableEvent) => {
       otherHandlers.onClick?.(event);
+      onItemClick?.(event, itemId);
```

Comment:
Another advantage of firing `onItemClick` here, people can easily use `event.defaultMuiPrevented` to prevent some built in logic.